File: internal/domain/game_event.go
package domain

import "time"

type GameEventType string

const (
	GameEventBumperHit  GameEventType = "bumper_hit"
	GameEventBallLost   GameEventType = "ball_lost"
	GameEventBonus      GameEventType = "bonus"
	GameEventFlipperHit GameEventType = "flipper_hit"
	GameEventGameOver   GameEventType = "game_over"
)

type GameEvent struct {
	ID        int64         `json:"id"`
	GameID    int64         `json:"game_id"`
	Type      GameEventType `json:"type"`
	Points    int64         `json:"points"`
	OccuredAt time.Time     `json:"occured_at"`
}

File: app/domain/game_event.py
from __future__ import annotations

from datetime import datetime
from enum import Enum

from pydantic import BaseModel


class GameEventType(str, Enum):
    GAME_STARTED = "game_started"
    BUMPER_HIT = "bumper_hit"
    BALL_LOST = "ball_lost"
    BONUS = "bonus"
    FLIPPER_HIT = "flipper_hit"
    GAME_OVER = "game_over"


class GameEvent(BaseModel):
    id: int | None = None
    game_id: int
    type: GameEventType
    points: int = 0
    occured_at: datetime = datetime.now()
